test/functional/backends/netssh_transfer_tests.rb

```diff
@@ -0,0 +1,83 @@
+require 'securerandom'
+
+module SSHKit
+  module Backend
+    module NetsshTransferTests
```

Comment:
🗒️ I extracted the upload/download tests into a module, so the same tests can easily be reused for both scp and sftp.